### ISSUE TITLE
very large instability

### ISSUE THREAD

**Issue by neda-noor-m** [TheShadow29/infnet-spen#1]
I ran your code on BiBTeX dataset and got the following plots. I was shocked, because these plots reflect very large instability. Did you get the same results on BiBTeX?  

![screenshot from 2018-11-27 12-31-59](https://user-images.githubusercontent.com/36059982/49070952-104c4f80-f242-11e8-9af8-12fa0e43839d.png)

**Comment by martiansideofthemoon** [comment]
Please specify the exact steps to reproduce this. (For eg, configuration file used) We did notice instability in the model and added the W-GAN extension to take care of it. Please see the report for more details.

**Comment by TheShadow29** [comment]
@neda-noor-m Can you turn off the ignore outliers in chart scaling (top left in tensorboard) and show the results again?

**Comment by neda-noor-m** [comment]
Thanks a lot for your prompt response.
I ran your code on BiBTeX with exactly same configuration mentioned in bibtex.json file. I didn't change anything. However, when I trained the code, I got the following F1 scores:
stage 1 - 0.409
stage 2 - 0.3932
I wonder why my results are not consistent with the reported results in Table 1.

**Comment by neda-noor-m** [comment]
When I turn off "ignore outliers in chart scaling"

![screenshot from 2018-11-28 10-20-38](https://user-images.githubusercontent.com/36059982/49134029-6c24e000-f2f7-11e8-8739-1af41b7cd299.png)

**Comment by TheShadow29** [comment]
I believe the graphs are similar to what we obtained as well, we too saw instability, and hence we proposed the wgan-style to avoid it.

I am not particularly sure why the F1-scores are lower though. I think they are close enough so I guess some tweaking of the hyper-params should help (some suggestions would be reduce the no. of epochs for stage 1 to say 20 or 30, and use learning rate of 1e-4 instead of 1e-3). 

@martiansideofthemoon  what do you think?

**Comment by martiansideofthemoon** [comment]
Curves are similar to what we report. I believe you are not reporting the early stopping figures / you haven't tuned the classification threshold on the development set. See 9.1 for reference (https://arxiv.org/pdf/1803.03376.pdf)

**Comment by neda-noor-m** [comment]
Thanks, I repeat the experiments using some tweaking of the hyper-params.
Meanwhile, I ran the wgan-style. Although the stability is better I got very bad and unexpected F1 scores as follow ("lamb_wgan_penalty": 1)!
stage 1 - 0.101
stage 2 - 0.2128
what do you think?

**Comment by neda-noor-m** [comment]
I got these plots with w-gan style. These plots are strange for me.

![screenshot from 2018-11-28 13-16-14](https://user-images.githubusercontent.com/36059982/49143208-cd58ad80-f30f-11e8-960b-af7389ec9b63.png)

**Comment by TheShadow29** [comment]
We got inferior performance with WGANs but it was small difference (1-2 F1-points I think). Do note that wgans need more iterations to converge, and their convergence was suboptimal. Even then if I remember correctly, the base objective should have reached to 1 or 2 rather than being around 100 so I am not sure what is going wrong.

As for the original work, I agree with @martiansideofthemoon. Tuning the threshold was quite important, and might be required to get good performance.

**Comment by neda-noor-m** [comment]
Thank you for your insightful comments
